src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueueDebugView.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Collections.Generic
+{
+    internal sealed class PriorityQueueDebugView<TElement, TPriority>
+    {
+        private readonly PriorityQueue<TElement, TPriority> _queue;
+        private readonly bool _sort;
+
+        public PriorityQueueDebugView(PriorityQueue<TElement, TPriority> queue)
+        {
+            _queue = queue ?? throw new ArgumentNullException(nameof(queue));
+            _sort = true;
+        }
+
+        public PriorityQueueDebugView(PriorityQueue<TElement, TPriority>.UnorderedItemsCollection collection)
+        {
+            _queue = collection?._queue ?? throw new ArgumentNullException(nameof(collection));
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public (TElement Element, TPriority Priority)[] Items
+        {
+            get
+            {
+                var list = new List<(TElement Element, TPriority Priority)>(_queue.UnorderedItems);
+                if (_sort)
+                {
+                    list.Sort((i1, i2) => _queue.Comparer.Compare(i1.Priority, i2.Priority));
```

Comment:
Simple test for this?
Also does this end up sorting high pri first?

Reply:
> does this end up sorting high pri first?

It's a min queue, and it correctly has lower priority values first:
![image](https://user-images.githubusercontent.com/2642209/110111070-c128e980-7d7d-11eb-95cc-0d535f619f40.png)

> Simple test for this?

Via reflection?

Reply:
Via compiling into test assy perhaps, but it’s not that important I guess